api/v1beta2/authpolicy_types.go

```diff
@@ -144,8 +144,18 @@ type CallbackSpec struct {
+// +kubebuilder:validation:XValidation:rule="self.targetRef.kind != 'Gateway' || !has(self.overrides) || !has(self.overrides.rules) || !has(self.overrides.rules.authentication) || !self.overrides.rules.authentication.exists(x, has(self.overrides.rules.authentication[x].routeSelectors))",message="route selectors not supported when targeting a Gateway"
+// +kubebuilder:validation:XValidation:rule="self.targetRef.kind != 'Gateway' || !has(self.overrides) || !has(self.overrides.rules) || !has(self.overrides.rules.metadata) || !self.overrides.rules.metadata.exists(x, has(self.overrides.rules.metadata[x].routeSelectors))",message="route selectors not supported when targeting a Gateway"
+// +kubebuilder:validation:XValidation:rule="self.targetRef.kind != 'Gateway' || !has(self.overrides) || !has(self.overrides.rules) || !has(self.overrides.rules.authorization) || !self.overrides.rules.authorization.exists(x, has(self.overrides.rules.authorization[x].routeSelectors))",message="route selectors not supported when targeting a Gateway"
+// +kubebuilder:validation:XValidation:rule="self.targetRef.kind != 'Gateway' || !has(self.overrides) || !has(self.overrides.rules) || !has(self.overrides.rules.response) || !has(self.overrides.rules.response.success) || !has(self.overrides.rules.response.success.headers) || !self.overrides.rules.response.success.headers.exists(x, has(self.overrides.rules.response.success.headers[x].routeSelectors))",message="route selectors not supported when targeting a Gateway"
+// +kubebuilder:validation:XValidation:rule="self.targetRef.kind != 'Gateway' || !has(self.overrides) || !has(self.overrides.rules) || !has(self.overrides.rules.response) || !has(self.overrides.rules.response.success) || !has(self.overrides.rules.response.success.dynamicMetadata) || !self.overrides.rules.response.success.dynamicMetadata.exists(x, has(self.overrides.rules.response.success.dynamicMetadata[x].routeSelectors))",message="route selectors not supported when targeting a Gateway"
+// +kubebuilder:validation:XValidation:rule="self.targetRef.kind != 'Gateway' || !has(self.overrides) || !has(self.overrides.rules) || !has(self.overrides.rules.callbacks) || !self.overrides.rules.callbacks.exists(x, has(self.overrides.rules.callbacks[x].routeSelectors))",message="route selectors not supported when targeting a Gateway"
 // Mutual Exclusivity Validation
 // +kubebuilder:validation:XValidation:rule="!(has(self.defaults) && (has(self.routeSelectors) || has(self.patterns) || has(self.when) || has(self.rules)))",message="Implicit and explicit defaults are mutually exclusive"
+// +kubebuilder:validation:XValidation:rule="!(has(self.overrides) && (has(self.routeSelectors) || has(self.patterns) || has(self.when) || has(self.rules)))",message="Implicit defaults and explicit overrides are mutually exclusive"
+// +kubebuilder:validation:XValidation:rule="!(has(self.overrides) && has(self.defaults))",message="Explicit overrides and explicit defaults are mutually exclusive"
```

Comment:
I'm not too sure this makes sense for `AuthPolicy` even if we only support atomic default/overrides at the moment, since the `AuthPolicyCommonSpec` has multiple fields aside from auth rules. For example, perhaps  user setting `defaults.routeselectors` and `override.rules` set valid, where inheriting policies will default to a some routerselctor but must use inherited auth rules from gateway policy  🤔  

Anyhow, just a thought, since if we keep it as is, the implementation logic is probably simpler

Reply:
This might be a misunderstanding. It was my understanding that in the first implementation was to set defaults and overrides for the spec and not the individual components with in the spec. The diagrams in the [RFC](https://github.com/Kuadrant/architecture/blob/main/rfcs/0009-defaults-and-overrides.md#examples-c---override-policy-entirely-replacing-other-at-lower-level) lead me to believe it was at the spec level. 

If I was wrong in my understanding then that changes a lot about how this implementation would work.

Reply:
> It was my understanding that in the first implementation was to set defaults and overrides for the spec and not the individual components with in the spec.

This is correct. Which means there's very little use case for `defaults` and `overrides` in a same policy. At least until we introduce merges, conditions and/or start allowing multiple policies of a kind to target a same resource, I suppose users will be limited to either defaults or overrides set at the level of a Gateway.

Reply:
Cool, thanks 👍 In that case, please disregard my comment